dataRetriever/resolvers/topicResolverSender/topicResolverSender.go

```diff
@@ -76,48 +85,54 @@ func (trs *topicResolverSender) SendOnRequestTopic(rd *dataRetriever.RequestData
 		return err
 	}
 
-	peerList := trs.peerListCreator.PeerList()
-	if len(peerList) == 0 {
-		return fmt.Errorf("%w, topic: %s", dataRetriever.ErrNoConnectedPeerToSendRequest, trs.topicName)
+	topicToSendRequest := trs.topicName + topicRequestSuffix
+
+	crossPeers := trs.peerListCreator.PeerList()
+	numSent := trs.sendOnTopic(crossPeers, topicToSendRequest, buff, trs.numCrossShardPeers)
+
+	intraPeers := trs.peerListCreator.IntraShardPeerList()
+	numSent = numSent + trs.sendOnTopic(intraPeers, topicToSendRequest, buff, trs.numIntraShardPeers)
+
+	if numSent == 0 {
+		return fmt.Errorf("%w or peer send error, topic: %s", dataRetriever.ErrSendRequest, trs.topicName)
```

Comment:
be careful when you wrap this error after you refactor (if you refactor) it the way I suggested

Reply:
right, simplified the wrap